client/rest/src/main/java/org/elasticsearch/client/ResponseException.java

```diff
@@ -33,7 +33,7 @@
 
     private Response response;
 
-    ResponseException(Response response) throws IOException {
+    public ResponseException(Response response) throws IOException {
```

Comment:
For consistent code, it helps a lot to actually be able to throw a `ResponseException` to get it handled the same way as other failures. The way that the client decides whether to throw one isn't generally as complete as most actual uses will need (e.g., perhaps `201` is an exception). On the other hand, mocking exceptions can be a bit silly (though the way that this build's the message somewhat makes me want to allow it, but not at the expense of making it extensible).